src/templates/service/service.template.ts

```diff
@@ -0,0 +1,97 @@
+import { methodNames } from '@utils/constants';
+import { ControllerPath } from '../controller/controller.template';
+import * as _ from 'lodash';
+
+
+type addServiceOperationProps = ControllerPath & {
+    serviceName: string,
+    imports?: any,
+}
+
+const getImportsAndClass = (serviceName: string, imports: any) => {
+
+    const cServiceName = _.capitalize(serviceName)
+
+    const dtosImports = [...imports.dtos].join(', ')
+    const entitiesImports = [...imports.entities].join(', ')
+
+    return `
+import { Injectable } from '@nestjs/common'
+// import { InjectRepository } from '@nestjs/typeorm'
+
+import {
+${dtosImports}
+} from './${serviceName}.dto'
+
+//import {
+//${entitiesImports}
+//} from './${serviceName}.entity'
+// import { Repository } from 'typeorm'
+
+@Injectable()
+export class ${cServiceName}Service {
+  constructor(
+    // @InjectRepository(_Category)
+    // private categoriesRepository: Repository<Category>
+  ) {}
+
+  `
+}
+
+
+const addServiceOperation = ({ method, pathParams, queryParams, body, returnType, imports }: addServiceOperationProps) => {
+
+    if (body) {
+        imports.dtos.add(body)
+    }
+
+    if (returnType) {
+        imports.dtos.add(_.capitalize(returnType.includes('[]') ? returnType.slice(0, returnType.length - 2) : returnType))
+    }
+
+    const methodName = `${methodNames[method.toUpperCase() as keyof typeof methodNames]}${pathParams ? `By${pathParams.map(_.capitalize).join()}` : ''}`
+
+    const pathParamsArgs = pathParams?.map(param => `${param}: string`).join(', ') ?? ''
+    const queryParamsArgs = queryParams?.map(param => `${param}: string`).join(', ') ?? ''
+    const bodyParamsArgs = body ? `body: ${body}` : ''
+
+    const argsParams = _.compact([pathParamsArgs, queryParamsArgs, bodyParamsArgs]).join(', ')
+
+    let returnObj: string
+    if (returnType?.includes('[]')) {
+        returnObj = '[]'
+    } else if (returnType) {
+        returnObj = `new ${returnType}()`
+    }
+    else {
+        returnObj = ''
+    }
+
+    return `
+        async ${methodName}(${argsParams}): Promise<${returnType ?? 'void'}> {
+            // template generated method
+            // update it as needed
+            return new Promise( (resolve, reject) => { return ${returnObj} } );
```

Comment:
Нет resolve в теле промиса, значит он в теории не должен отработать. 

```suggestion
            return new Promise( (resolve, reject) => { return ${returnObj} } );
```

Reply:
>  ${returnObj} не в бэктиках (`)
этот блок кода - возвращаемый многострочный текст https://developer.mozilla.org/ru/docs/Learn/JavaScript/First_steps/Strings#%D0%BC%D0%BD%D0%BE%D0%B3%D0%BE%D1%81%D1%82%D1%80%D0%BE%D1%87%D0%BD%D1%8B%D0%B9_%D1%82%D0%B5%D0%BA%D1%81%D1%82
кавычки есть на той же строке где return и на следующей строке после `return new Promise`
![Uploading image.png…]()
оборачивать еще раз бэктиками `returnObj` не нужно